pkg/util/vault.go

```diff
@@ -59,47 +60,44 @@ func NewVaultClient(pi *PodInfo) (*vaultapi.Client, error) {
 		return nil, errors.Wrap(err, "failed to unmarshal parameters")
 	}
 
-	binding := app.DeepCopy()
-	var sar *core.ObjectReference
-
-	if cf.UsePodServiceAccountForCSIDriver {
-		// Use the JWT token of pod's service account
-		// to perform Kubernetes auth in the Vault server.
-		cf.ServiceAccountName = pi.ServiceAccount
-		binding.Spec.Secret = nil
-
-		// Get pod's service account
-		sa, err := pi.KubeClient.CoreV1().ServiceAccounts(pi.Namespace).Get(pi.ServiceAccount, metav1.GetOptions{})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to get pod's service account")
+	// If Kubernetes authentication information is given,
+	// perform Kubernetes auth to the Vault servers.
+	if cf.Kubernetes != nil {
+		authInfo := &authtype.AuthInfo{
+			VaultApp: app,
 		}
 
-		sar = &core.ObjectReference{
-			Name:      sa.Name,
-			Namespace: sa.Namespace,
-		}
-		// Get the role name from service account annotations.
-		// Kubernetes authentication will be performed in the Vault server against this role.
-		if pbRole, ok := sa.Annotations[PolicyBindingRole]; ok {
-			cf.PolicyControllerRole = pbRole
+		if cf.Kubernetes.UsePodServiceAccountForCSIDriver {
+			// Get pod's service account
+			sa, err := pi.KubeClient.CoreV1().ServiceAccounts(pi.Namespace).Get(pi.ServiceAccount, metav1.GetOptions{})
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get pod's service account")
+			}
+			authInfo.ServiceAccountRef = &core.ObjectReference{
+				Name:      sa.Name,
+				Namespace: sa.Namespace,
+			}
+
+			// Get the role name from service account annotations.
+			// Kubernetes authentication will be performed in the Vault server against this role.
+			if pbRole, ok := sa.Annotations[VaultRole]; ok {
+				authInfo.VaultRole = pbRole
+			} else {
+				return nil, errors.New("failed to get policy binding role from pod's service account")
+			}
 		} else {
-			return nil, errors.New("failed to get policy binding role from pod's service account")
+			authInfo.ServiceAccountRef = &core.ObjectReference{
+				Name:      cf.Kubernetes.ServiceAccountName,
```

Comment:
Check cf.Kubernetes != nil

Reply:
This part of code is already inside of an `if cf.Kubernetes != nil` block which started at line 65.